monai/data/box_utils.py

```diff
@@ -932,3 +938,78 @@ def box_pair_giou(boxes1: NdarrayOrTensor, boxes2: NdarrayOrTensor) -> NdarrayOr
     # convert tensor back to numpy if needed
     giou, *_ = convert_to_dst_type(src=giou_t, dst=boxes1)
     return giou
+
+
+def spatial_crop_boxes(
+    boxes: NdarrayOrTensor,
+    roi_start: Union[Sequence[int], NdarrayOrTensor],
+    roi_end: Union[Sequence[int], NdarrayOrTensor],
+    remove_empty: bool = True,
+) -> Tuple[NdarrayOrTensor, NdarrayOrTensor]:
+    """
+    This function generate the new boxes when the corresponding image is cropped to the given ROI.
+    When ``remove_empty=True``, it makes sure the bounding boxes are within the new cropped image.
+
+    Args:
+        boxes: bounding boxes, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+        roi_start: voxel coordinates for start of the crop ROI, negative values allowed.
+        roi_end: voxel coordinates for end of the crop ROI, negative values allowed.
+        remove_empty: whether to remove the boxes that are actually empty
+
+    Returns:
+        - cropped boxes, boxes[keep], does not share memory with original boxes
```

Comment:
Could you please help add a test to ensure the returned `cropped boxes` doesn't share memory with original input boxes?

Thanks.